dipy/tracking/utils.py

```diff
@@ -397,21 +379,17 @@ def seeds_from_mask(mask, density=[1, 1, 1], voxel_size=None, affine=None):
     seeds = seeds.reshape((-1, 3))
 
     # Apply the spatial transform
-    if affine is not None:
+    if seeds.any():
```

Comment:
I do no understand this Condition

Reply:
Before the affine was optional, so it had to be tested if it existed.
Now it is not need (since it is mandatory)

However I realize that a test supposed to check if empty seed mask were crashing was not actully getting tested because there was no affine provided and so was not getting transform.

So to apply a transform to seed, you need at least one (thats why any() is used, any non zero value in the array)

Reply:
Thank you for the explanation